chat_stream.go

```diff
@@ -48,12 +51,19 @@ func (stream *ChatCompletionStream) Recv() (response ChatCompletionStreamRespons
 waitForData:
 	line, err := stream.reader.ReadBytes('\n')
 	if err != nil {
+		if errRes, _ := stream.errAccumulator.unmarshalError(); errRes != nil {
+			err = fmt.Errorf("error, %w", errRes.Error)
+		}
 		return
 	}
 
 	var headerData = []byte("data: ")
 	line = bytes.TrimSpace(line)
 	if !bytes.HasPrefix(line, headerData) {
+		if writeErr := stream.errAccumulator.write(line); writeErr != nil {
```

Comment:
We can cover this with tests if we make `errorAccumulator` an interface and provide failing one like here https://github.com/sashabaranov/go-openai/blob/aa149c1bf8ccd77403e07fdb7b0122f27d4c9b6b/request_builder_test.go#L30